pkg/apis/maistra/conversion/conversion_test.go

```diff
@@ -275,6 +276,604 @@ var (
 				},
 			}),
 		},
+		{
+			name: "MAISTRA-1983.1",
+			smcpv1: v1.ControlPlaneSpec{
+				Version: "v1.1",
+				Istio: v1.NewHelmValues(map[string]interface{}{
+					"tracing": map[string]interface{}{
+						"jaeger": map[string]interface{}{
+							"elasticsearch": map[string]interface{}{
+								"nodeCount":        int64(5),
+								"redundancyPolicy": nil,
+								"esIndexCleaner": map[string]interface{}{
+									"enabled":      true,
+									"numberOfDays": int64(60),
+									"schedule":     "55 23 * * *",
+								},
+							},
+						},
+					},
+				}),
+			},
+			smcpv2: v2.ControlPlaneSpec{
+				Version: "v1.1",
+				Addons: &v2.AddonsConfig{
+					Jaeger: &v2.JaegerAddonConfig{
+						Install: &v2.JaegerInstallConfig{
+							Storage: &v2.JaegerStorageConfig{
+								Elasticsearch: &v2.JaegerElasticsearchStorageConfig{
+									NodeCount: &jaegerElasticsearchNodeCount,
+								},
+							},
+						},
+					},
+				},
+				TechPreview: v1.NewHelmValues(map[string]interface{}{
+					"tracing": map[string]interface{}{
+						"jaeger": map[string]interface{}{
+							"elasticsearch": map[string]interface{}{
+								"esIndexCleaner": map[string]interface{}{
+									"enabled":      true,
+									"numberOfDays": int64(60),
+									"schedule":     "55 23 * * *",
+								},
+							},
+						},
+					},
+				}),
+			},
+			roundTripped: &v1.ControlPlaneSpec{
+				Version: "v1.1",
+				Istio: v1.NewHelmValues(map[string]interface{}{
+					"tracing": map[string]interface{}{
+						"jaeger": map[string]interface{}{
+							"elasticsearch": map[string]interface{}{
+								"nodeCount": int64(5),
+								"esIndexCleaner": map[string]interface{}{
+									"enabled":      true,
+									"numberOfDays": int64(60),
+									"schedule":     "55 23 * * *",
+								},
+							},
+						},
+					},
+				}),
+			},
+			cruft: v1.NewHelmValues(map[string]interface{}{
+				"global": map[string]interface{}{
+					// mesh expansion is disabled by default
+					"meshExpansion": globalMeshExpansionDefaults,
+					// multicluster is disabled by default
+					"multiCluster": globalMultiClusterDefaults,
+				},
+			}),
+		},
+		{
+			name: "MAISTRA-1983.2",
+			smcpv1: v1.ControlPlaneSpec{
+				Version: "v1.1",
+				Istio: v1.NewHelmValues(map[string]interface{}{
+					"tracing": map[string]interface{}{
+						"jaeger": map[string]interface{}{
+							"elasticsearch": map[string]interface{}{
+								"nodeCount":        int64(5),
+								"redundancyPolicy": nil,
```

Comment:
is there a reason you set this `nil` specifically?

Reply:
nevermind I just tested with different string values and it is preserved

Reply:
Yes, this was the issue fixed in MAISTRA-1983.  `unstructured.NestedString()` returns an error if the field is nil.  Personally, I think this is a bug in apimachinery, because if the parent field is nil, or the field doesn't exist at all, the call succeeds, but returns not found.  I suppose you could argue that the field was found and it's not a string, so...